azuredevops/internal/service/policy/common.go

```diff
@@ -63,7 +63,59 @@ func genBasePolicyResource(crudArgs *policyCrudArgs) *schema.Resource {
 		Update:   genPolicyUpdateFunc(crudArgs),
 		Delete:   genPolicyDeleteFunc(crudArgs),
 		Importer: tfhelper.ImportProjectQualifiedResourceInteger(),
-		Schema:   genBaseSchema(),
+		Schema: map[string]*schema.Schema{
+			SchemaProjectID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			SchemaEnabled: {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			SchemaBlocking: {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			SchemaSettings: {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						SchemaScope: {
+							Type: schema.TypeList,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									SchemaRepositoryID: {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									SchemaRepositoryRef: {
```

Comment:
Are `SchemaRepositoryID` and `SchemaRepositoryRef` related in any way? Are they mutual exclusive or something?

Reply:
I really don't know..

Reply:
Perhaps you can dig this up and add one of the available schema constraints `ConflictsWith`, `ExactlyOneOf`, `AtLeastOneOf` and `RequiredWith` . Where I suspect that `RequiredWith` makes any sense here.

Reply:
![image](https://user-images.githubusercontent.com/12761692/87411710-ffd81c80-c5bf-11ea-8c85-ba8dafd354da.png)
It's the branch

Reply:
Ah I see 👍 Then it's a perfect application for the `RequiredWith` constrained on `repository_ref` because I reckon that you can't use `repository_ref` **without** `repository_id`.

Reply:
![image](https://user-images.githubusercontent.com/12761692/87416968-63b21380-c5c7-11ea-97b3-56a8abe3d23f.png)
I am not so sure, but I still can't tell

@xuzhang3 Do you have any idea?

Reply:
If I'm not wrong, `repository_ref ` contains the git commit logs.  @tmeckel  is right here, can't use `repository_ref ` without `repository_id`

Reply:
Then the `RequiredWith:  []string{ "repository_id" }, ` constraint should be added to the `repository_ref` schema definition.

Reply:
Done